Modules/Core/Common/include/itkImageRegionConstIteratorWithIndex.hxx

```diff
@@ -32,8 +32,8 @@ ImageRegionConstIteratorWithIndex<TImage>::operator++()
   this->m_Remaining = false;
   for (unsigned int in = 0; in < TImage::ImageDimension; in++)
   {
-    this->m_PositionIndex[in]++;
-    if (this->m_PositionIndex[in] < this->m_EndIndex[in])
+    const IndexValueType positionIndex = this->m_PositionIndex[in]++;
```

Comment:
Consider: 
const IndexValueType positionIndex = ++this->m_PositionIndex[in];

Reply:
Good catch